internal/handler/handler.go

```diff
@@ -139,6 +151,69 @@ func New(opts *NewOpts) (*Handler, error) {
 	return h, nil
 }
 
+// Setup creates initial database and user if needed.
+func (h *Handler) setup() error {
+	if h.SetupDatabase == "" {
+		return nil
+	}
+
+	ctx, cancel := context.WithTimeout(context.TODO(), h.SetupTimeout)
+	defer cancel()
+
+	info := conninfo.New()
+	info.SetBypassBackendAuth()
+
+	ctx = conninfo.Ctx(ctx, info)
+
+	l := h.L.Named("setup")
+
+	var retry int64
+
+	for ctx.Err() == nil {
+		_, err := h.b.Status(ctx, nil)
+		if err == nil {
+			break
+		}
+
+		l.Debug("Status failed", zap.Error(err))
+
+		retry++
+		ctxutil.SleepWithJitter(ctx, time.Second, retry)
+	}
+
+	res, err := h.b.ListDatabases(ctx, &backends.ListDatabasesParams{Name: h.SetupDatabase})
+	if err != nil {
+		return err
+	}
+
+	if len(res.Databases) > 0 {
+		l.Debug("Database already exists")
```

Comment:
If the database exists but the user doesn't exist, we don't create user here?

Reply:
The idea is to do nothing if some initialization was already performed before